lib/auth/auth_with_roles.go

```diff
@@ -1622,6 +1622,15 @@ func (a *ServerWithRoles) ListResources(ctx context.Context, req proto.ListResou
 		}()
 	}
 
+	// ensure the 'SSHTargetHost' field's matching behavior is consistent with the routing
+	// configuration in cluster networking config.
+	req.CaseInsensitiveRouting = false
+	if req.SSHTargetHost != "" {
+		if cfg, err := a.authServer.GetClusterNetworkingConfig(ctx); err == nil {
+			req.CaseInsensitiveRouting = cfg.GetCaseInsensitiveRouting()
+		}
+	}
```

Comment:
Do we not always have at least some default `ClusterNetworkingConfig`, where `CaseInsensitiveRouting` defaults to `false?
```suggestion
		if req.SSHTargetHost != "" {
		cfg, err := a.authServer.GetClusterNetworkingConfig(ctx);
		if err != nil {
			return nil, trace.Wrap(err)
		}
		req.CaseInsensitiveRouting = cfg.GetCaseInsensitiveRouting()
	}
```

Even if not, the lack of error handling smells off.

Reply:
I don't disagree really, but we have an established pattern for routing config where proxies fallback to using the old/default routing behavior if they can't load `ClusterNetworkingConfig`.  In the interest of keeping things persistent, I prefer the idea of having the `SSHTargetHost` param make decisions as close to how the actual proxy routing logic makes decisions as possible, so I'm thinking I'll leave this as-is.

Reply:
I don't quite get it, but I'll trust your judgement as long as you've double checked.